Java 9 rt.jar export incompatible with -jvm-debug option: "java.lang.ClassNotFoundException: for"

```
⚡ (java_use 9; bash -x /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt -jvm-debug 5006)
++ is_cygwin
+++ uname -s
++ local os=Darwin
++ case "$os" in
++ return 1
++ echo false
+ CYGWIN_FLAG=false
+++ realpath /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt
+++ TARGET_FILE=/usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt
+++ FIX_CYGPATH=
++++ dirname /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt
+++ cd /usr/local/Cellar/sbt/1.0.0/libexec/bin
++++ basename /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt
+++ TARGET_FILE=sbt
+++ COUNT=0
+++ '[' -L sbt -a 0 -lt 100 ']'
+++ [[ x != \x ]]
++++ pwd -P
+++ echo /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt
++ dirname /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt
+ . /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch-lib.bash
++ declare -a residual_args
++ declare -a java_args
++ declare -a scalac_args
++ declare -a sbt_commands
++ declare java_cmd=java
++ declare java_version
++ declare init_sbt_version=1.0.0
++ declare SCRIPT=/usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt
++ '[' -h /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt ']'
+++ dirname /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt
++ declare -r sbt_bin_dir=/usr/local/Cellar/sbt/1.0.0/libexec/bin
+++ dirname /usr/local/Cellar/sbt/1.0.0/libexec/bin
++ declare -r sbt_home=/usr/local/Cellar/sbt/1.0.0/libexec
+ declare -r 'noshare_opts=-Dsbt.global.base=project/.sbtboot -Dsbt.boot.directory=project/.boot -Dsbt.ivy.home=project/.ivy'
+ declare -r sbt_opts_file=.sbtopts
+ declare -r etc_sbt_opts_file=/usr/local/etc/sbtopts
+ declare -r win_sbt_opts_file=/usr/local/Cellar/sbt/1.0.0/libexec/conf/sbtconfig.txt
+ [[ -f /usr/local/etc/sbtopts ]]
++ loadConfigFile /usr/local/etc/sbtopts
++ cat /usr/local/etc/sbtopts
++ sed '/^\#/d'
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo -sbt-create
+++ echo -sbt-create
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ eval echo
+++ echo
++ read -r line
++ [[ -n '' ]]
+ set -- -sbt-create -jvm-debug 5006
+ [[ -f .sbtopts ]]
+ [[ -f .jvmopts ]]
+ run -sbt-create -jvm-debug 5006
+ syncPreloaded
+ [[ 1.0.0 == '' ]]
+ [[ -f /Users/jz/.sbt/preloaded/org.scala-sbt/sbt/1.0.0/jars/sbt.jar ]]
+ [[ -f '' ]]
+ acquire_sbt_jar ''
++ jar_file
+++ cygwinpath /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch.jar
+++ local file=/usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch.jar
+++ [[ false == \t\r\u\e ]]
+++ echo /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch.jar
++ echo /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch.jar
+ sbt_jar=/usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch.jar
+ [[ ! -f /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch.jar ]]
+ process_args -sbt-create -jvm-debug 5006
+ [[ 3 -gt 0 ]]
+ case "$1" in
+ addResidual -sbt-create
+ dlog '[residual] arg = '\''-sbt-create'\'''
+ [[ -n '' ]]
+ residual_args=("${residual_args[@]}" "$1")
+ shift
+ [[ 2 -gt 0 ]]
+ case "$1" in
+ require_arg port -jvm-debug 5006
+ local type=port
+ local opt=-jvm-debug
+ local arg=5006
+ [[ -z 5006 ]]
+ [[ 5 == \- ]]
+ addDebugger 5006
+ addJava -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5006
+ dlog '[addJava] arg = '\''-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5006'\'''
+ [[ -n '' ]]
+ java_args=("${java_args[@]}" "$1")
+ shift 2
+ [[ 0 -gt 0 ]]
+ is_function_defined process_my_args
+ declare -f process_my_args
+ myargs=("${residual_args[@]}")
+ residual_args=()
+ process_my_args -sbt-create
+ [[ 1 -gt 0 ]]
+ case "$1" in
+ sbt_create=true
+ shift
+ [[ 0 -gt 0 ]]
+ [[ XXX != \X\X\X ]]
++ java -Xmx512M -version
++ grep ' version "'
++ sed 's/.*version "\([0-9]*\)\(\.[0-9]*\)\{0,1\}\(.*\)*"/\1\2/; 1q'
+ java_version=9
+ vlog '[process_args] java_version = '\''9'\'''
+ [[ -n '' ]]
+ [[ -n '' ]]
+ set --
+ argumentCount=0
+ checkJava 1.6
+ local required_version=1.6
+ [[ 9 == '' ]]
+ [[ 9 < 1.6 ]]
+ copyRt
+ [[ 9 == \9 ]]
++ rt_export_file
++ echo /usr/local/Cellar/sbt/1.0.0/libexec/bin/java9-rt-export.jar
+ rtexport=/usr/local/Cellar/sbt/1.0.0/libexec/bin/java9-rt-export.jar
++ java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5006 -jar /usr/local/Cellar/sbt/1.0.0/libexec/bin/java9-rt-export.jar --rt-ext-dir
+ java9_ext='Listening for transport dt_socket at address: 5006
/Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9'
++ echo 'Listening for transport dt_socket at address: 5006
/Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9/rt.jar'
+ java9_rt='Listening for transport dt_socket at address: 5006
/Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9/rt.jar'
+ vlog '[copyRt] java9_rt = '\''Listening for transport dt_socket at address: 5006
/Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9/rt.jar'\'''
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ ! -f Listening for transport dt_socket at address: 5006
/Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9/rt.jar ]]
+ addJava '-Dscala.ext.dirs=Listening for transport dt_socket at address: 5006
/Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9'
+ dlog '[addJava] arg = '\''-Dscala.ext.dirs=Listening for transport dt_socket at address: 5006
/Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9'\'''
+ [[ -n '' ]]
+ java_args=("${java_args[@]}" "$1")
+ [[ false == \t\r\u\e ]]
++ get_mem_opts
++ [[ '' == *-Xmx* ]]
++ [[ '' == *-Xms* ]]
++ [[ '' == *-XX:MaxPermSize* ]]
++ [[ '' == *-XX:MaxMetaspaceSize* ]]
++ [[ '' == *-XX:ReservedCodeCacheSize* ]]
++ [[ '' == *-Xmx* ]]
++ [[ '' == *-Xms* ]]
++ [[ '' == *-XX:MaxPermSize* ]]
++ [[ '' == *-XX:MaxMetaspaceSize* ]]
++ [[ '' == *-XX:ReservedCodeCacheSize* ]]
++ [[ -Xss3M -Xmx2G == *-Xmx* ]]
++ echo ''
+ execRunner java -Xss3M -Xmx2G -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5006 -Dscala.ext.dirs=Listening for transport dt_socket at address: 5006 /Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9 -jar /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch.jar
+ [[ -n '' ]]
+ [[ -n '' ]]
+ java -Xss3M -Xmx2G -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5006 -Dscala.ext.dirs=Listening for transport dt_socket at address: 5006 /Users/jz/.sbt/0.13/java9-rt-ext-oracle_corporation_9 -jar /usr/local/Cellar/sbt/1.0.0/libexec/bin/sbt-launch.jar
Listening for transport dt_socket at address: 5006
Error: Could not find or load main class for
Caused by: java.lang.ClassNotFoundException: for
+ exit_code=1
+ [[ false == \t\r\u\e ]]
+ exit 1
```

--- a/src/universal/bin/sbt-launch-lib.bash
+++ b/src/universal/bin/sbt-launch-lib.bash
@@ -229,7 +229,7 @@ copyRt() {
   if [[ "$at_least_9" == "1" ]]; then
     rtexport=$(rt_export_file)
     java9_ext=$("$java_cmd" ${JAVA_OPTS} ${SBT_OPTS:-$default_sbt_opts} ${java_args[@]} \
-      -jar "$rtexport" --rt-ext-dir)
+      -jar "$rtexport" --rt-ext-dir | grep -v Listening)
     java9_rt=$(echo "$java9_ext/rt.jar")
     vlog "[copyRt] java9_rt = '$java9_rt'"
     if [[ ! -f "$java9_rt" ]]; then

